accelerometer/summariseEpoch.py

```diff
@@ -218,11 +219,14 @@ def check_daylight_savings_crossover(e, startTime, endTime, summary,
         for t in localTime._utc_transition_times:
             if t>startTime:
                 transition = t
+                transition = transition - pd.DateOffset(seconds = 1)
```

Comment:
The key challenge in this work is that all timestamps in e.index are defined without timezone information, meaning that when a timestamp is defined as 5pm, it's supposed to refer to 5pm local time, however the timestamp does not have timezone association, in effect in "UTC"

However, the output of pytz._utc_transition_times are the dst crossover times defined in UTC, for example, while most dst crossover happens at 2am, the "America/New_York" (UTC-5) 's _utc_transition_times are 7am during spring crossover, and 6am during fall crossover. To address this, a series of transformation is required on the output of _utc_transition_times, the variable "transition" here.

The first step is a "hack" that dial the "transition" back for 1 second. This is because, during spring crossover, the same UTC transition time, 7am (in UTC) for US eastern, corresponds to both 2am and 3am local time, as they are essentially the same time. By dialing back 1 second, 6:59:59am UTC will deterministically become 1:59:59am US/eastern.